doc/examples/examples.hpp.in

```diff
@@ -0,0 +1,276 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2019, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+
+/**
+ * @page Examples Example programs
+ *
+ * Here you can find example programs that demonstrate the usage of Ginkgo.
+ * Some examples are built on one another and some are stand-alone and
+ * demonstrate a concept of
+ * Ginkgo, which can be used in your own code.
+ *
+ * You can browse the available example programs
+ * <ol>
+ *   <li> as <b><a href="#graph">a graph</a></b> that shows how example programs
+ *     build upon each other.
+ *   <li> as <b><a href="#list">a list</a></b> that provides a short
+ *     synopsis of each program.
+ *   <li> or <b><a href="#topic">grouped by topic</a></b>.
+ * </ol>
+ *
+ * By default, all Ginkgo examples are built using CMake.
+ *
+ * An example for building the examples and using Ginkgo as an external library
+ * without CMake can be found in the script provided for each example, which
+ * should be called with the form:
+ * <code>./build.sh PATH_TO_GINKGO_BUILD_DIR </code>
+ *
+ * By default, Ginkgo is compiled with at least
+ * <code>-DGINKGO_BUILD_REFERENCE=on</code>.
```

Comment:
I would prefer to always use capitalized `ON` and `OFF`, which we do most of the time.
```suggestion
 * <code>-DGINKGO_BUILD_REFERENCE=ON</code>.
```